databricks/koalas/generic.py

```diff
@@ -1281,6 +1281,9 @@ def groupby(self, by, as_index: bool = True):
             raise ValueError("Grouper for '{}' not 1-dimensional".format(type(by)))
         if not len(by):
             raise ValueError('No group keys passed!')
+        if not isinstance(as_index, bool):
+            raise TypeError('as_index must be an boolean; however, '
```

Comment:
nit: `as_index must be an boolean` -> `as_index must be a boolean`

Reply:
hmm, maybe this way looks fine for now, but i think maybe we should handle `by` parameter directly rather than `as_index` since now behavior looks some hacky (e.g. if other parameters will be added to second positional parameter, it will not work properly (maybe `axis`, `level` like the pandas are doing that like the below)

```python
    @Appender(_shared_docs["groupby"] % _shared_doc_kwargs)
    def groupby(
        self,
        by=None,
        axis=0,
        level=None,
        as_index: bool = True,
        sort: bool = True,
        group_keys: bool = True,
        squeeze: bool = False,
        observed: bool = False,
    ) -> "groupby_generic.DataFrameGroupBy":
```

Reply:
Oppose~ I made a mistake. I'll fix this.

Reply:
i think we can handle axis parameter like pandas.

```python
ValueError: No axis named A for object type <class 'pandas.core.frame.DataFrame'>
```

Reply:
> hmm, maybe this way looks fine for now, but i think we better handle `by` parameter directly rather than `as_index` since now behavior looks some hacky (e.g. if other parameters will be added to second positional parameter, it will not work properly - maybe `axis` or `level` like the below pandas are doing-)
> 
> ```python
>     @Appender(_shared_docs["groupby"] % _shared_doc_kwargs)
>     def groupby(
>         self,
>         by=None,
>         axis=0,
>         level=None,
>         as_index: bool = True,
>         sort: bool = True,
>         group_keys: bool = True,
>         squeeze: bool = False,
>         observed: bool = False,
>     ) -> "groupby_generic.DataFrameGroupBy":
> ```

Pandas recognized the value of the second argument as an "axis". Koalas does not yet support "axis" parameters, so it is recognized as "as_index". It is expected that "as_index" will be safely verified no matter what function is added in the future.

Reply:
> i think we can handle `axis` parameter like pandas.
> 
> ```python
> ValueError: No axis named A for object type <class 'pandas.core.frame.DataFrame'>
> ```
> 
> for example, you can add `axis` parameter with default 0, and raise `NotImplementedError` for when `axis=1` for now.


Let me proceed that way. Thank you :)

Reply:
yes, go for it!

FYI: although i think your approach is not bad, but the point is that we always try to mimic pandas as possible as we can 😃

Reply:
And pandas doesn't raise exception when `as_index` is not boolean.

- for string

```python
>>> pdf.groupby(['A', 'B'], as_index='koalas').sum()
                  C         D
A   B
bar one   -0.517176  1.082738
    three  2.075121  0.745619
    two    1.062240 -0.289503
foo one    0.028217 -0.131834
    three -0.374145 -0.422457
    two    2.320670  1.416018
```

- for number

```python
>>> pdf.groupby(['A', 'B'], as_index=100).sum()
                  C         D
A   B
bar one   -0.517176  1.082738
    three  2.075121  0.745619
    two    1.062240 -0.289503
foo one    0.028217 -0.131834
    three -0.374145 -0.422457
    two    2.320670  1.416018
```

but we are 

```python
>>> kdf.groupby('A', as_index='koalas').sum()
Traceback (most recent call last):
...
TypeError: as_index must be an boolean; however, got [<class 'str'>]
```

Reply:
> And pandas doesn't raise exception when `as_index` is not boolean.
> (They treat 0 as False otherwise True)
> 
> * `True` for string or number which is not 0.
> 
> ```python
> >>> pdf.groupby('A', as_index='koalas').sum()
>             C         D
> A
> bar -0.998532  1.623860
> foo  3.844849  1.563355
> 
> >>> pdf.groupby('A', as_index=100).sum()
>             C         D
> A
> bar -0.998532  1.623860
> foo  3.844849  1.563355
> ```
> 
> * `False` for 0
> 
> ```python
> >>> pdf.groupby('A', as_index=0).sum()
>      A         C         D
> 0  bar -0.998532  1.623860
> 1  foo  3.844849  1.563355
> ```
> 
> but we're not.
> 
> ```python
> >>> kdf.groupby('A', as_index='koalas').sum()
> Traceback (most recent call last):
> ...
> TypeError: as_index must be an boolean; however, got [<class 'str'>]
> ```

I got it! I will remove the "as_index" validation to work with the example shown above.